### PR TITLE
MCPWM deadtime improvements

### PR DESCRIPTION
**Pull request by dimpolo**
* allow changing deadtime config after pin creation
* constify DeadTimeCfg
* fix some wrong bit values
* remove some unsafe blocks

### CODE REVIEW

File: esp-hal/src/mcpwm/operator.rs
Comment (MabezDev) [inline esp-hal/src/mcpwm/operator.rs:344]: It might be helpful to state why this could be bad, and what users can do to avoid that bad behaviour when updating the dead time?

File: esp-hal/CHANGELOG.md
Comment (MabezDev) [inline esp-hal/CHANGELOG.md:34]: I think you meant MCPWM in these changelog entries right?

**Review by MabezDev — CHANGES_REQUESTED** [review]
Thanks for the PR, I still need to do some testing on hardware, so bear with me.

This is optional, but do you think adding deadtime configuration to the existing example or a new one would be a good idea?

File: esp-hal/CHANGELOG.md
Comment (dimpolo) [reply esp-hal/CHANGELOG.md:34]: Yes!

File: esp-hal/src/mcpwm/operator.rs
Comment (dimpolo) [reply esp-hal/src/mcpwm/operator.rs:344]: Or maybe something like:

```rust
impl Operator {
    pub fn with_linked_pins(self, pin_1: Pin1, config_a: PwmPinConfig, pin_2: Pin2, config_b: PwmPinConfig, config_dt: DeadTimeConfig) -> LinkedPins;
}

pub struct LinkedPins;

impl LinkedPins {
    pub fn set_actions_a(&mut self, value: PwmActions);
    pub fn set_actions_b(&mut self, value: PwmActions);
    pub fn set_update_method_a(&mut self, update_method: PwmUpdateMethod);
    pub fn set_update_method_b(&mut self, update_method: PwmUpdateMethod);
    pub fn set_timestamp_a(&mut self, value: u16);
    pub fn set_timestamp_b(&mut self, value: u16);
    pub fn set_deadtime_config(&mut self, config: DeadTimeConfig);
    pub fn set_rising_edge_deadtime(&mut self, dead_time: u16);
    pub fn set_falling_edge_deadtime(&mut self, dead_time: u16);
}
```

Comment (MabezDev) [reply esp-hal/src/mcpwm/operator.rs:344]: I think the latter could work, but I will admit my knowledge of MCPWM is a bit limited, I need to do some reading :D. Maybe we could discuss this in a separate issue? I think the API for MCPWM could do with a bit of work regardless, but I don't want to block this PR.

**Comment by dimpolo** [comment]
@MabezDev Ready for review

**Comment by jessebraham** [comment]
Thank you for your work on this!

Most of our maintainers are traveling this week, but we will get to this ASAP! If @MabezDev (or somebody else) does not get to this in the next couple days I will give it a review/test later this week, but I will be traveling over the next couple days as well.

**Comment by dimpolo** [comment]
All good! I hope all of you enjoy your travels 😁

File: esp-hal/src/mcpwm/operator.rs
Comment (MabezDev) [inline esp-hal/src/mcpwm/operator.rs:486]:
```suggestion
/// let bridge_off = DeadTimeCfg::new_bypass().set_output_swap(PWMStream::PWMA, true);
```

**Review by MabezDev — CHANGES_REQUESTED** [review]
Thank you so much for the PR! Apologies again for the delay. This turned out nicely, and I really appreciate the effort you put into the doc improvements too! It's much appreciated.

I have the smallest nit-pick ever (sorry!), but you should be able to commit directly via GH, then we'll get this merged :).

Edit: turns out I can apply the suggestion too!

**Review by MabezDev — APPROVED** [review]
Oops, I forgot to actually approve this :sweat_smile:

**Review by bjoernQ — APPROVED** [review]

**Comment by MabezDev** [comment]
Ugh, I'm so sorry @dimpolo would you mind rebasing? Any errors related to `variant()` should now use `bits()` instead.

**Comment by jessebraham** [comment]
Sorry I missed the bit in @MabezDev's comment about `variant`, was going to rebase this (hence the change to the `CHANGELOG.md`) but I will not do that now 😁  Sorry for the noise!

**Comment by dimpolo** [comment]
Sorry, I made a mess of the commit log 😅 

> Any errors related to variant() should now use bits() instead.

Oh wow, that's unfortunate. Reading through the logs there seems to be a new `set` method to replace `variant` but it's only available for `Safe` fields.
https://github.com/rust-embedded/svd2rust/pull/829
Maybe our PACs can be updated to enable this in the future

**Pull request merged**
